admin/model/extension/module/altapay.php

```diff
@@ -53,5 +63,44 @@ public function updateOrderMeta($order_id, $capture = false, $refund = false, $v
             $this->db->query("UPDATE " . DB_PREFIX . "altapay_orders SET modified='".$this->db->escape((string)date('Y-m-d H:i:s', time()))."', $status_query WHERE order_id='".(int)$order_id."'");
         }
 	}
-	
+
+    /**
+     * @param int $order_id
+     *
+     * @return array
+     */
+   public function getOrderReconciliationIdentifiers($order_id)
+    {
+        $query = $this->db->query('SELECT reconciliation_identifier, transaction_type FROM `' . DB_PREFIX . 'altapay_order_reconciliation` WHERE order_id =' . (int) $order_id);
+
+        if ($query->num_rows) {
+            return $query->rows;
+        } else {
+            return [];
+        }
+    }
+
+    /**
+     * @param int $order_id
+     * @param string $reconciliation_identifier
+     * @param string $type
+     *
+     * @return void
+     */
+    public function saveOrderReconciliationIdentifier($order_id, $reconciliation_identifier, $type = 'captured') {
+        $query = $this->db->query(
+            'SELECT id FROM `' . DB_PREFIX . 'altapay_order_reconciliation` WHERE order_id =' . (int) $order_id .
+            " AND reconciliation_identifier ='" . $this->db->escape((string) $reconciliation_identifier) .
+            "' AND transaction_type ='" . $this->db->escape((string) $type) . "'");
+
+        if (!$query->num_rows) {
+            $this->db->query(
+                'INSERT INTO `' . DB_PREFIX . 'altapay_order_reconciliation` 
+                (order_id, reconciliation_identifier, transaction_type) 
+                VALUES ' . "('" . (int) $order_id . "', 
```

Comment:
Casting a string will give you a `0` as a result, causing side effects for unvalidated order ids

Reply:
I've updated such that the SQL will only execute if `order_id` passes integer validation > 0.